test/unit/gateways/reach_test.rb

```diff
@@ -130,6 +129,33 @@ def test_sucess_from_on_failure
     refute @gateway.send(:success_from, response)
   end
 
+  def test_stored_credential
+    @options[:stored_credential] = { initiator: 'cardholder', initial_transaction: true, reason_type: 'unschedule' }
```

Comment:
💬  Comment:

Maybe this is a good scenario to test a combination of values and their expected output so you can have a lot of combinations covered in the test without creating a test for each transactions, something like this https://gist.github.com/54e9832c74bce33de4e84a49b8b85e29